File: services/_template-ts/src/health.ts
import type { Router } from 'express';
import { Router as createRouter } from 'express';

let ready = false;

export function setReady(value: boolean): void {
  ready = value;
}

export function healthRouter(): Router {
  const router = createRouter();

  router.get('/healthz', (_req, res) => {
    res.status(200).json({ status: 'ok' });
  });

  router.get('/readyz', (_req, res) => {
    if (ready) {
      res.status(200).json({ status: 'ready' });
    } else {
      res.status(503).json({ status: 'not_ready' });
    }
  });

  return router;
}

File: services/telemetry/src/health.py
from __future__ import annotations

from fastapi import APIRouter, Response, status

router = APIRouter()

_ready = False


def set_ready(value: bool) -> None:
    global _ready
    _ready = value


@router.get("/healthz")
async def healthz() -> dict[str, str]:
    return {"status": "ok"}


@router.get("/readyz")
async def readyz(response: Response) -> dict[str, str]:
    if not _ready:
        response.status_code = status.HTTP_503_SERVICE_UNAVAILABLE
        return {"status": "not_ready"}
    return {"status": "ready"}
